src/imgui/ImGuiReverseBar.cc

```diff
@@ -94,7 +132,7 @@ void ImGuiReverseBar::showMenu(MSXMotherBoard* motherBoard)
 						if (name.ends_with(STATE_EXTENSION)) {
 							name.remove_suffix(STATE_EXTENSION.size());
 							std::filesystem::file_time_type ftime = std::filesystem::last_write_time(fullName);
-							stateNames.emplace_back(std::string(name), ftime);
+							stateNames.emplace_back("", std::string(name), ftime); // we don't care about full name
```

Comment:
There's no harm in filling in `fullName` anyway. That would allow to factor out a function to scan all files in a directory (you pass `STATE_DIR` and `STATE_EXTENSION` as parameters).

And having the fullname available is actually useful:
* Then searching for the corresponding `.png` file becomes simpler (just replace `.oms` with `.png`).
* I haven't checked, but maybe the `loadstate` command accepts full names (that save a lookup during execution of that command). By itself maybe not a useful tweak, but it again makes the savestate and replay code more similar.